### PR TITLE
Update `make docs` procedure and add workflow to keep it up to date

### PR DESCRIPTION
**Pull request by jdbaldry**
#### What this PR does

Updates the `make docs` procedure, consisting of the script and related Make scaffolding.

**For reviewers:**
1. Please ensure you can run `make docs` from the `docs/` directory.
1. Update repository access, please do one of the following:
   1. Add @grafana/docs-tooling with write access to the repository. It is a group containing only myself but would additionally include any other Doc squad tooling maintainers in the future. Any changes to this groups membership would be communicated.
    1. Add only my GitHub handle if that is easier for governance purposes. 
    1. Ask me to remove the changes to CODEOWNERS and rely on my membership of the @grafana/mimir-maintainers group.

### CODE REVIEW

File: docs/make-docs [hunk 1]
```diff
@@ -478,6 +490,8 @@ POSIX_HERESTRING
 
   echo
   errr 'The build was interrupted or a build error occurred, check the previous logs for possible causes.'
+  note 'You might need to use Ctrl+c to end the process.'
```
Comment (osg-grafana) [inline docs/make-docs:493]:
```suggestion
  note 'You might need to use Ctrl+C to end the process.'
```
For keys, use WYSIWYG. If we were to use shift, we would write Ctrl-Shift-C.

**Review by osg-grafana — APPROVED** [review]
Approving from a docs side with feedback; please have an engineer vet this for technical accuracy.

File: docs/make-docs (hunk 1 above)
Comment (jdbaldry) [reply docs/make-docs:493]: Makes sense, I can make this change upstream and it will get propagated to all other repositories.

**Comment by jdbaldry** [comment]
@grafana/mimir-maintainers Anyone able to give this a look?

File: .github/workflows/update-make-docs.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Update `make docs` procedure
+on:
+  schedule:
+    - cron: '0 7 * * 1-5'
+  workflow_dispatch:
+jobs:
+  main:
+    if: github.repository == 'grafana/opentelemetry-docs'
```
Comment (charleskorn) [inline .github/workflows/update-make-docs.yml:8]: Should this be `grafana/mimir`?

**Comment by charleskorn** [comment]
> Update repository access, please do one of the following:
>   1. Add @grafana/docs-tooling with write access to the repository. It is a group containing only myself but would additionally include any other Doc squad tooling maintainers in the future. Any changes to this groups membership would be communicated.
>   2. Add only my GitHub handle if that is easier for governance purposes.
>   3. Ask me to remove the changes to CODEOWNERS and rely on my membership of the @grafana/mimir-maintainers group.

I'm OK with the first option, will leave this for a day or two for other opinions.

File: .github/workflows/update-make-docs.yml (hunk 2 above)
Comment (jdbaldry) [reply .github/workflows/update-make-docs.yml:8]: Good catch, thank you.

I was cleaning up a bunch of old PRs like this and forgot to update the copy-paste in this one.

```suggestion
    if: github.repository == 'grafana/mimir'
```

**Review by charleskorn — APPROVED** [review]
LGTM.

I've given @grafana/docs-tooling write access to this repo.

**Comment by jdbaldry** [comment]
Cheers @charleskorn!

**Pull request merged**